psh/pshapp/pshapp.c

```diff
@@ -1096,18 +1096,29 @@ static int psh_runscript(char *path)
 		return -EINVAL;
 	}
 
-	free(line);
-	line = NULL;
-	n = 0;
+	for (i = 2;; i++) {
+		ssize_t lineLen = getline(&line, &n, stream);
+		if (lineLen <= 0) {
+			break;
+		}
+
+		if (line[lineLen - 1] == '\n') {
+			lineLen--;
+			line[lineLen] = '\0';
+		}
```

Comment:
Consider moving the `SCRIPT_MAGIC` check to the for loop:
```suggestion
	for (i = 1;; i++) {
		ssize_t lineLen = getline(&line, &n, stream);
		if (lineLen <= 0) {
			break;
		}

		if (line[lineLen - 1] == '\n') {
			lineLen--;
			line[lineLen] = '\0';
		}

		if ((i == 1) && (strcmp(line, SCRIPT_MAGIC) != 0)) {
			fprintf(stderr, "psh: %s is not a psh script\n", path);
			free(line);
			fclose(stream);
			return -EINVAL;
		}
```

Reply:
Thanks for the suggestion, for now I will stay with the current version, where SCRIPT_MAGIC is checked before the loop.

Reply:
But the current version allows such a line:
```
:{}:123
```

Reply:
You're right, I couldn't write it more optimally (I mean checking before the loop, because MAGIC is actually checked only once) without duplicating the code. I accept your suggestion. Thanks for spotting this.